osbenchmark/workload_generator/workload_generator.py

```diff
@@ -123,36 +135,84 @@ def create_workload(cfg):
 
     logger.info("Creating workload [%s] matching indices [%s]", workload_name, indices)
     logger.info("Number of Docs: %s", number_of_docs)
-    client = OsClientFactory(hosts=target_hosts.all_hosts[opts.TargetHosts.DEFAULT],
-                             client_options=client_options.all_client_options[opts.TargetHosts.DEFAULT]).create()
+    client = OsClientFactory(
+        hosts=target_hosts.all_hosts[opts.TargetHosts.DEFAULT],
+        client_options=client_options.all_client_options[opts.TargetHosts.DEFAULT],
+    ).create()
 
     info = client.info()
-    console.info(f"Connected to OpenSearch cluster [{info['name']}] version [{info['version']['number']}].\n", logger=logger)
+    console.info(
+        f"Connected to OpenSearch cluster [{info['name']}] version [{info['version']['number']}].\n",
+        logger=logger,
+    )
+
+    output_path = os.path.abspath(
+        os.path.join(io.normalize_path(root_path), workload_name)
+    )
+    operations_path = os.path.join(output_path, "operations")
+    test_procedures_path = os.path.join(output_path, "test_procedures")
 
-    output_path = os.path.abspath(os.path.join(io.normalize_path(root_path), workload_name))
     io.ensure_dir(output_path)
+    io.ensure_dir(operations_path)
+    io.ensure_dir(test_procedures_path)
 
-    indices, corpora = extract_mappings_and_corpora(client, output_path, indices, number_of_docs)
+    indices, corpora = extract_mappings_and_corpora(
+        client, output_path, indices, number_of_docs
+    )
 
+    # user logger.info and print indices and corpora
     if len(indices) == 0:
         raise RuntimeError("Failed to extract any indices for workload!")
 
     template_vars = {
         "workload_name": workload_name,
         "indices": indices,
         "corpora": corpora,
-        "custom_queries": custom_queries
+        "custom_queries": custom_queries,
     }
 
     logger.info("Template Vars: %s", template_vars)
 
     workload_path = os.path.join(output_path, "workload.json")
+    operations_path = os.path.join(operations_path, "default.json")
+    test_procedures_path = os.path.join(test_procedures_path, "default.json")
     templates_path = os.path.join(cfg.opts("node", "benchmark.root"), "resources")
 
+    write_template(
+        workload_path,
+        extract_template(templates_path, "base-workload.json.j2"),
+        template_vars,
+    )
+
     if custom_queries:
```

Comment:
Recommend moving this to a single function in the same module to reduce boilerplate code.